From 5d4106eafd7f2f6e71a1d0d90ebf4f4e766b772d Mon Sep 17 00:00:00 2001
From: Assaf Gordon <assafgordon@gmail.com>
Date: Mon, 15 Sep 2014 19:34:06 +0000
Subject: [PATCH 12/26] virt-what.in: make option processing portable

Replace linux-specific 'getopt' usage with manual option parsing.
Because the used option are version simple (help/version) and the only
additional option is undocumented (--test-root), it's easy remove the
'getopt' usage, and make this script portable to BSD systems.
---
 virt-what.in | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/virt-what.in b/virt-what.in
index 6e9aef1..f2c0b90 100644
--- a/virt-what.in
+++ b/virt-what.in
@@ -51,24 +51,21 @@ usage() {
 }
 
 # Handle the command line arguments, if any.
-
-TEMP=$(getopt -o v --long help --long version --long test-root: -n 'virt-what' -- "$@")
-if [ $? != 0 ]; then exit 1; fi
-eval set -- "$TEMP"
-
-while true; do
+while test $# -gt 0; do
     case "$1" in
 	--help) usage ;;
-        --test-root)
+        --test-root=*)
             # Deliberately undocumented: used for 'make check'.
-            root="$2"
-            shift 2
+            root=$(echo "$1" | sed 's/.*=//')
+            shift 1
+            test -z "$root" && fail "--test-root option requires a value"
             ;;
 	-v|--version) echo "$VERSION"; exit 0 ;;
 	--) shift; break ;;
-	*) fail "internal error ($1)" ;;
+	*) fail "unrecognized option '$1'";;
     esac
 done
+test $# -gt 0 && fail "extra operand '$1'"
 
 # Add /sbin and /usr/sbin to the path so we can find system
 # binaries like dmidecode.
-- 
1.8.3.1

